addons/addon-base-raas/packages/base-raas-services/lib/data-source/access-strategy/roles-only/__tests__/application-role-methods.test.js

```diff
@@ -0,0 +1,338 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+// Mocked services
+const {
+  toAppRoleEntity,
+  toDbEntity,
+  newAppRoleEntity,
+  addStudy,
+  maxReached,
+  toCfnResources,
+} = require('../helpers/entities/application-role-methods');
+
+const createStudy = ({
+  id = 'study-1',
+  category = 'Organization',
+  accountId = '1122334455',
+  awsPartition = 'aws',
+  bucketAccess = 'roles',
+  kmsArn = undefined,
+  bucket = 'bucket-1',
+  qualifier = 'swb-IhsKhN8GsLneiis11ujlb8',
+  appRoleArn = 'arn:aws:iam::123456789012:role/swb-IhsKhN8GsLneiis11ujlb8-app-1234567890xxx',
+  accessType = 'readwrite',
+  envPermission = { read: true, write: true },
+  folder = '/',
+  kmsScope = 'none',
+} = {}) => ({
+  id,
+  category,
+  accountId,
+  awsPartition,
+  bucketAccess,
+  bucket,
+  qualifier,
+  appRoleArn,
+  kmsArn,
+  accessType,
+  envPermission,
+  folder,
+  kmsScope,
+});
+
+const createAppRole = ({
+  arn = 'arn:aws:iam::123456789012:role/swb-IhsKhN8GsLneiis11ujlb8-app-1234567890xxx',
+  accountId = '1122334455',
+  mainRegion = 'us-east-1',
+  awsPartition = 'aws',
+  bucket = 'bucket-1',
+  by = 'sampleUser',
+  bucketRegion = 'us-east-1',
+  status = 'pending',
+  name = 'swb-IhsKhN8GsLneiis11ujlb8-app-1234567890xxx',
+  qualifier = 'swb-IhsKhN8GsLneiis11ujlb8',
+  boundaryPolicyArn = 'arn:aws:iam::123456789012:policy/swb-IhsKhN8GsLneiis11ujlb8-app-1234567890xxx',
+  studies = {
+    'study-1': {
+      accessType: 'readonly',
+      kmsScope: 'none',
+      folder: '/',
+    },
+  },
+} = {}) => ({
+  accountId,
+  arn,
+  name,
+  mainRegion,
+  by,
+  awsPartition,
+  bucket,
+  bucketRegion,
+  status,
+  qualifier,
+  boundaryPolicyArn,
+  studies,
+});
+
+describe('ApplicationRoleMethods', () => {
+  beforeEach(async () => {});
+
+  it('ensures toAppRoleEntity does not fail with undefined dbEntity', async () => {
```

Comment:
Can we add describe clause for different methods? It'll be easier to maintain and update.